src/hsm.cpp

```diff
@@ -104,6 +130,12 @@ openssl::SSL_EVP_PKEY_Ptr HsmEngine::generateKey(const RSASpec &spec,
                                                  const std::string &keyLabel,
                                                  const std::vector<uint8_t> &keyID)
 {
+    // Check for duplicate key generation:
+    checkForDuplicateKey(keyLabel, keyID);
+    checkForDuplicateKey("", keyID);
```

Comment:
What does this do? Does it search for the empty label or does it not include the label at all?